lib/pafs_core/data_migration/remove_previous_years.rb

```diff
@@ -1,35 +1,43 @@
 # frozen_string_literal: true
 
-require "csv"
-
 module PafsCore
   module DataMigration
     class RemovePreviousYears
-      attr_reader :project, :current_year
 
-      def self.perform_all
-        PafsCore::Project.find_each do |project|
-          new(project).perform
+      def self.perform(max_projects = 100)
+        processed_project_count = 0
+
+        # The funding_values cleanup for a given project will also clear out any previous
+        # years data for the other annualised attributes. We run the service for each of
+        # those attributes to handle any cases where previous years funding_values were not
+        # present but previous years data was present for another attribute.
+        %i[funding_values
+           flood_protection_outcomes
+           flood_protection2040_outcomes
+           coastal_erosion_protection_outcomes].each do |attribute|
+          projects = qualifying_projects(attribute, max_projects - processed_project_count)
+          puts "Processing #{projects.length}/#{max_projects} projects based on #{attribute}" unless Rails.env.test?
+          projects.each do |project|
+            PafsCore::RemovePreviousYearsService.new(project).run
+          end
+
+          processed_project_count += projects.length
+          break if processed_project_count >= max_projects
         end
       end
 
-      def initialize(project)
-        @project = project
+      def self.qualifying_projects(attribute, project_limit)
+        PafsCore::Project
+          .joins(:state)
+          .where.not(pafs_core_states: { state: :submitted })
```

Comment:
Not sure if it makes sense to delete previous years data for archived projects. We'll end up with plenty of empty archived projects that have no years data. 

It would make sense in case we replace deleted years records with a summary record.

Reply:
It will only affect two areas in the project, funding years and properties affected years (OM4s I think they are called)
You can change the status from draft to archived (and back) so I think this is ok. As they aren't submitted at this point users will get an opportunity to enter data for the current and future years when in draft, so I think this is ok.